File: Basics/Conditional-Statements-Advanced/Lab/fruit_shop.py
fruit = input().lower()
day = input()
quantity = float(input())
week = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]
weekend = ["Saturday", "Sunday"]
fruits = ["banana", "apple", "orange", "grapefruit", "kiwi", "pineapple", "grapes"]
price = 0
is_fruit_valid = False
is_day_valid = False
if fruit in fruits:
    is_fruit_valid = True
if day in week:
    is_day_valid = True

    if fruit == "banana":
        price = 2.50
    elif fruit == "apple":
        price = 1.20
    elif fruit == "orange":
        price = 0.85
    elif fruit == "grapefruit":
        price = 1.45
    elif fruit == "kiwi":
        price = 2.70
    elif fruit == "pineapple":
        price = 5.50
    elif fruit == "grapes":
        price = 3.85
elif day in weekend:
    is_day_valid = True

    if fruit == "banana":
        price = 2.70
    elif fruit == "apple":
        price = 1.25
    elif fruit == "orange":
        price = 0.90
    elif fruit == "grapefruit":
        price = 1.60
    elif fruit == "kiwi":
        price = 3
    elif fruit == "pineapple":
        price = 5.60
    elif fruit == "grapes":
        price = 4.20

if is_fruit_valid and is_day_valid:
    end_sum = quantity * price
    print(f"{end_sum:.2f}")
else:
    print(f"error")

File: JavaScript/Basics/0.3-Conditional-Steatements-Advanced/Lab/fruit-shop.js
function fruitShop(input) {
    let fruit, day, quantity;
    [fruit, day, quantity] = input;
    quantity = Number(quantity);
    let week = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"];
    let workingDays = {
        "banana": 2.50,
        "apple": 1.20,
        "orange": 0.85,
        "grapefruit": 1.45,
        "kiwi": 2.70,
        "pineapple": 5.50,
        "grapes": 3.85
    }
    let weekends = {
        "banana": 2.70,
        "apple": 1.25,
        "orange": 0.90,
        "grapefruit": 1.60,
        "kiwi": 3,
        "pineapple": 5.60,
        "grapes": 4.20
    }

    if (day == "Sunday" || day == "Saturday") {
        if (fruit in weekends) {
            console.log((quantity * weekends[fruit]).toFixed(2));
        } else {
            console.log("error");
        }
    } else if (week.includes(day)) {
        if (fruit in workingDays) {
            console.log((quantity * workingDays[fruit]).toFixed(2));
        } else {
            console.log("error");
        }
    } else {
        console.log("error");
    }
}

fruitShop(["apple", "Tuesday", "2"])
fruitShop(["orange", "Sunday", "3"])
fruitShop(["kiwi", "Monday", "2.5"])
fruitShop(["grapes", "Saturday", "0.5"])
fruitShop(["tomato", "Monday", "0.5"])